packages/util/src/internal.ts

```diff
@@ -135,10 +135,13 @@ export function toAscii(hex: string): string {
  * @param optional padding
  * @returns hex representation of input string
  */
-export function fromUtf8(stringValue: string) {
-  const str = utf8ToBytes(stringValue)
+export function fromUtf8(utf8String: string): string {
+  if (typeof utf8String !== 'string') {
+    throw new Error(`[fromUtf8] input must be a string, received type ${typeof utf8String}`)
+  }
 
-  return `0x${padToEven(bytesToHex(str)).replace(/^0+|0+$/g, '')}`
```

Comment:
I'm not sure what the original regex did, remove preceding 0s and trailing 0s? Can't we use the original and instead use `bytesToUnprefixedHex` instead of `bytesToHex`?

Reply:
Yes, I've changed over to this.